Adjust Colors for the Solidity Docs

4molybdenum2 opened the pull request:
Made the following changes in this PR  (#11900): 

- [x] Add custom CSS file called "light.css" with the color changes as follows:
- Make code block background Solidity light grey: #E5E5E5
- Make titles / highlights Solidity light blue: #65AFFF
- Make hyperlinks Solidity electric blue: #002fa7
- Make highlights for searched words warmer yellow: #EAC545

- [x] In the dark mode CSS file, adjust the search highlight color to be soft red: #F0978B (yellow is currently not readable)
- [x] [optional] Add Solidity logo in the menu bar and delete logo from the index page
- [ ] [optional] Adjust font to Work Sans to be consistent with the font across blog, portal and docs

4molybdenum2 committed:
Adjust Colors for the Solidity Docs

ekpyron commented:
The CI failure is independent of the PR and can be ignored (that's just CircleCI acting up again).

As for the changed colors: one thing I'm noticing is that the new darker blue for links makes the links quite hard to read if switched to dark (using the toggle in the bottom left).

4molybdenum2 commented:
So should I change it back to the original color currently or maybe a different color for dark mode? 🤔

ekpyron commented:
Switching back to the old link color only in dark mode would seem fine with me, but I'm not a designer, so I'd defer that question to you or @franzihei :-).

4molybdenum2 commented:
Sure👍

franzihei commented:
Good point I did not think about the hyperlink being too dark in dark mode. I think keeping the original hyperlink color in the dark mode would work great!

franzihei commented on docs/_static/css/light.css:
```diff
@@ -0,0 +1,30 @@
+/* inline code highlights */
+.xref,
+.py-meth,
+.rst-content code {
```
Here I meant to change the color of the code blocks, as in the background color in the example contracts (currently a greenish yellow), not the inline code highlights. Sorry if that wasn't clear!

4molybdenum2 replied:
Oh ok, I will fix these things and make a commit then..

4molybdenum2 replied:
Was unable to resolve the **version issue where 'v' was missing**. Seems to be existing. I think we need to pass the version as v0.8.8 in the version variable (which I think is the **CMakeLists.txt**)  file.

franzihei reviewed (commented):
Thanks so much for picking up this issue! 

A couple of things:
- In the dark version now all text in the menu sidebar seems to be light blue. This shouldn't be the case, it should remain in the style it was before
Now:
![image](https://user-images.githubusercontent.com/41991517/132882535-b369c50a-c594-40fb-bef4-34d1a4946b6b.png)
Before:
![image](https://user-images.githubusercontent.com/41991517/132882574-0b437d63-976a-4526-8103-db9bd924ccb3.png)

- The background of the code blocks (the big boxes with code) should be light grey, not the background of inline code.
- In the header below the logo it now only says `0.8.8` as opposed to `v0.8.8`. The `v`seems to be missing.
- Is it possible to make the logo a bit smaller/ have less free space around it?

franzihei reviewed (commented):
Thanks for the changes! One thing that is not optimal are the "visited" hyperlinks in the menu on the left. The menu items should not change their color to light blue when they were clicked/visited, they should keep the same color (that applies for both light mode and dark mode).

Also I think that simply making the logo smaller does not work, it looks like there is too much "empty space" surrounding it. That might be an issue with the original image used. 

![image](https://user-images.githubusercontent.com/41991517/133090052-4605c0e3-8296-42e5-9317-1021a36e0115.png)

4molybdenum2 commented:
Hmm, the thing that I couldn't figure out was how to remove the "empty space" around the symbol because the space is in the **png** and we can't remove that with CSS itself.

4molybdenum2 commented:
Hmm...I couldn't figure out the "empty space" scenario.

cameel commented on docs/_static/css/light.css:
```diff
@@ -0,0 +1,31 @@
+
+.wy-side-nav-search img{
+	width: 80px !important;
+}
```
I think this should be more specific. We only want this height specifically for the logo:
```suggestion
.wy-side-nav-search img.logo {
	width: 80px !important;
}
```

cameel replied:
Also, why is this in `light.css`? Don't we need this rule in dark mode as well?

cameel commented on docs/conf.py:
```diff
@@ -142,7 +145,7 @@ def setup(sphinx):
 # The name of an image file (relative to this directory) to place at the top
 # of the sidebar.
-#html_logo = None
+html_logo = "./logo.svg"
```
```suggestion
html_logo = "logo.svg"
```

cameel commented on docs/conf.py:
```diff
@@ -154,7 +157,7 @@ def setup(sphinx):
 html_static_path = ['_static']
 
-html_css_files = ["css/toggle.css"]
+html_css_files = ["css/toggle.css", "css/light.css"]
```
Won't this apply `light.css` in both modes rather than only in the light mode as the name would suggest?

If your additions there were meant for both modes then I think that `custom.css` would be a better place.

cameel commented on docs/conf.py:
```diff
@@ -124,11 +124,14 @@ def setup(sphinx):
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 html_theme = 'sphinx_rtd_theme'
-
```
I'd leave this line in. The file separates options with empty lines for readability.

cameel commented on docs/conf.py:
```diff
@@ -124,11 +124,14 @@ def setup(sphinx):
+html_theme_options = {
+    'logo_only': True,
+    'style_nav_header_background': '#65afff',
+    'display_version': True,
```
> Was unable to resolve the **version issue where 'v' was missing**. Seems to be existing. I think we need to pass the version as v0.8.8 in the version variable (which I think is the **CMakeLists.txt**)  file.

I think that version gets overridden by ReadTheDocs. If you build docs locally on develop you don't get the `v` either. And ReadTheDocs can also display something completely different there - for it's `latest` for https://docs.soliditylang.org/en/latest/ and `breaking` for https://docs.soliditylang.org/en/breaking/ even though these do have 0.8.8 in `CMakeLists.txt`.

I think that the `v` will be there when the PR gets merged and docs get deployed.

cameel commented on docs/_static/css/dark.css:
```diff
@@ -46,7 +45,9 @@ em.property {
 }
 
-
+.rst-content pre{
+    background: none;
+}
```
```suggestion
.rst-content pre {
    background: none;
}

```

cameel commented on docs/_static/css/light.css:
```diff
@@ -0,0 +1,31 @@
+/* links */
+.rst-content a{
+    color: #002fa7;
+}
+
+.rst-content a:visited {
+    color: #002fa7;
+}
```
I think the visited color was meant to be identical to normal color only in the menu. Currently links in the text do get highlighted when visited and I'd personally count that as a feature :)

> One thing that is not optimal are the "visited" hyperlinks in the menu on the left. The menu items should not change their color to light blue when they were clicked/visited, they should keep the same color (that applies for both light mode and dark mode).

BTW, I can understand why designers might not like menu items having different colors when visited but personally I find that feature very useful so I'd keep the visited colors even in the menu. The light blue was definitely too flashy but maybe we could have at least a subtle color change? What do you think @franzihei?

cameel commented on docs/_static/css/light.css:
```diff
@@ -0,0 +1,31 @@
+	color: #272525 !important;
+}
+
+/* title color */
```
"title color" sounds as if these were titles in the text
```suggestion
/* menu section headers */
```

cameel replied:
@4molybdenum2 Can you fix this one too?

cameel commented on docs/_static/css/light.css:
```diff
@@ -0,0 +1,31 @@
+	width: 80px !important;
+}
+
+/* version sidebar */
```
```suggestion
/* project version (displayed under project logo) */
```

cameel commented on docs/_static/css/light.css:
```diff
@@ -0,0 +1,31 @@
+/* links */
+.rst-content a{
```
```suggestion
.rst-content a {
```

cameel commented on docs/_static/css/light.css:
```diff
@@ -0,0 +1,31 @@
+	background: #e5e5e5;
+}
+
+.wy-side-nav-search img{
+	width: 80px !important;
+}
```
> Hmm...I couldn't figure out the "empty space" scenario.

- The `<a>` element containing the logo has a bottom margin of `0.809em` that could be removed.
- Then the `<div>` that contains that link has `0.809em` of padding.
- Something adds padding of `5px` around `.wy-side-nav-search img` - if you remove it, logo gets slightly bigger.
- `.wy-side-nav-search > div.version` has a negative top margin. If the above are not enough, you could use that to get version as close to the logo as needed.

I think that the above bring us pretty close to the spacing in the design.

cameel commented on docs/_static/css/dark.css:
```diff
@@ -70,6 +71,12 @@ code.docutils.literal.notranslate {
 
+/* highlight color search text */
+
+.rst-content .highlighted {
+    background: #f0978b;
+    box-shadow: 0 0 0 2px #f0978b;
+}
```
> In the dark mode CSS file, adjust the search highlight color to be soft red: #F0978B (yellow is currently not readable)

This is more readable now but I think it could still be improved. The contrast between the text and the highlighted background seems a bit low so it might still be hard to read in some conditions:

![contract-metadata-dark-mode](https://user-images.githubusercontent.com/137030/133288885-9957318a-a24c-424f-8394-391a320eeff3.png)

cameel commented on docs/_static/css/dark.css:
```diff
@@ -1,6 +1,5 @@
-
-a,
+.rst-content a,
 a:visited {
     color: #aaddff;
 }
```
This made links in page header and footer dark blue:

![contract-metadata-header-dark-mode](https://user-images.githubusercontent.com/137030/133289716-c438cf85-4656-4a0c-8078-368844e3d6f3.png)
![contract-metadata-footer-dark-mode](https://user-images.githubusercontent.com/137030/133290735-a446d4f8-19ba-42b7-809c-e0fdeaeda512.png)

cameel reviewed (dismissed):
Thanks for the PR @4molybdenum2! Franzi is off this week so here's some feedback from me. Looks like there are only tiny details to iron out at this point.

cameel commented on docs/_static/css/dark.css:
```diff
@@ -46,7 +45,9 @@ em.property {
 }
 
-
+.rst-content pre{
+    background: none;
+}
```
One more thing, I'd change the background here to white instead and make the border thicker (https://github.com/ethereum/solidity/issues/11900#issuecomment-919297759, https://github.com/ethereum/solidity/issues/11900#issuecomment-919311493).

cameel replied:
@axic What do you think? I think that green ones looked slightly better but white ones are not bad either so I'm fine with them.

4molybdenum2 commented:
Sure, I will get on it and make updates by tomorrow...

4molybdenum2 commented:
So, the changes I made acc. to review:
- refactored code wherever necessary
- updated the code blocks background to white and thickened the border
- made sure tho reduce space around logo as much as possible by removing the paddings and margins
- added a different color for the highlights in dark mode which provides better contrast
- deleted the light.css file and put everything in custom.css

cameel commented:
@4molybdenum2 Looks like all the visual issues have been resolved :) I think that the final result looks good. Now @franzihei just needs to approve :)

There's just some general cleanup needed before this can be merged:
- I see that `custom.css` has 2-space indents and you followed that style. Which is good overall, but the file is very short so maybe you could just convert it to 4 so that we have it consistent in all CSS files?
- There are some files like `toggle.css`, where the only change seems to be adding a newline at the end. Could you remove these unrelated changes from the PR?
- There are a few places with inconsistent whitespace, e.g. missing space before `{`, empty line inside `{}` or no empty line between CSS definitions. Could you fix those? I pointed some of them in comments above.
- Commits should be squashed. We generally perfer one logical change per commit so review fixes should be squashed in to the commits they fix rather than left on top.

4molybdenum2 commented:
Sorry, I have been away for a while and could not get to working on the above changes.....will do so whenever I get the time.

cameel commented:
@4molybdenum2 If you don't have time for it, I can take over the PR and finish it. You already did 99% of the work so it just needs some cleanup :)

4molybdenum2 commented:
> @4molybdenum2 If you don't have time for it, I can take over the PR and finish it. You already did 99% of the work so it just needs some cleanup :)

Sorry I couldn't get back to it. I am currently away for the time being 😅 .....I would really appreciate it if you could take over

cameel committed:
Consistent indentation in doc config, templates and static files

cameel commented:
I did the cleanup and also adjusted the link style so that visited link color is overridden only in the menu.

@franzihei Please take a look at this. Also, any feedback on https://github.com/ethereum/solidity/pull/11927#discussion_r708379674?

cameel commented:
@franzihei 
- Code block background changed to `#fafafa`.
- Border around code blocks was `0.01em` `solid` `#4a4a4a`. Tried to change it to `1px` `solid` `#cccccc` as on the blog but turns out that there's another border on the element the block is wrapped in (`.highlight-solidity`). The problem with overriding it is that I would have to do it for every language separately. Instead I just removed our border - the default one (`1px` `solid` `#e1e4e5`) looks pretty close to the one from the blog. Just has no rounded corners.
- About the `v`, see https://github.com/ethereum/solidity/pull/11927#discussion_r708373582. In short: the `v` should appear there once docs are deployed to Read The Docs.
- I don't see anything here or in the issue about what the color of the `v0.x.x` part was meant to be but it looks to me like it matches the colors in your design. It's currently `#272525`. Is that too black? I can change it to anything you want.

franzihei reviewed (dismissed):
Great, I think everything can stay like that! :)
